### PR TITLE
Prep for v1.23.4

### PR DESCRIPTION
**Pull request by odow**
## Pre-release

 - [x] Check that the pinned packages in `docs/Project.toml` are updated. We pin
       the versions so that changes in the solvers (changes in printing, small
       numeric changes) do not break the printing of the JuMP docs in arbitrary
       commits.
 - [x] Check that the `rev` fields in `docs/packages.toml` are updated. We pin
       the versions of solvers and extensions to ensure that changes to their
       READMEs do not break the JuMP docs in arbitrary commits, and to ensure
       that the versions are compatible with the latest JuMP and
       MathOptInterface releases.
 - [x] Check compat of `DimensionalData` in `Project.toml`
 - [x] Check compat of `MacroTools` in `Project.toml`
 - [x] Update `docs/src/changelog.md`
 - [x] https://github.com/jump-dev/JuMP.jl/actions/runs/11730588420
 - [x] Change the version number in `Project.toml`
 - [x] The commit messages in this PR do not contain `[ci skip]`

## The release

 - [ ] After merging this pull request, comment `[at]JuliaRegistrator register` in
       the GitHub commit. This should automatically publish a new version to the
       Julia registry, as well as create a tag, and rebuild the documentation
       for this tag.

       These steps can take quite a bit of time (1 hour or more), so don't be
       surprised if the new documentation takes a while to appear. In addition,
       the links in the README will be broken until JuliaHub fetches the new
       version on their servers.

## Post-release

 - [ ] Once the tag is created, update the relevant `release-` branch. The latest
       release branch at the time of writing is `release-1.0` (we haven't
       back-ported any patches that needed to create a `release-1.Y` branch). To
       to update the release branch with the v1.10.0 tag, do:
       ```
       git checkout release-1.0
       git pull
       git merge v1.10.0
       git push
       ```

### CODE REVIEW

File: docs/src/changelog.md [hunk 1]
```diff
@@ -7,6 +7,25 @@ CurrentModule = JuMP
...
+### Other
+
+ - Various minor improvements to the documentation (#3855) (#3860)
+ - Added [MathOptAI.jl](@ref) and [MathOptSymbolicAD.jl](@ref) (#3858)
+ - Clarified [`add_to_expression!`](@ref) can add two expressions (#3859)
+ - Added `SHOT` to the installation table (#3853)
+ - Improvements to test coverage (#3867) (#3868) (#3869) (#3870) (#3871) (#3872)
+   (#3873) (#3874) (#3875)
+ - JuMP now uses [`MOI.add_constrained_variable`](@ref) when adding a scalar
+   variable with bounds (#3863) (#3865)
```
Comment (odow) [inline docs/src/changelog.md:27]: This seemed like a `MINOR` feature change, but it doesn't actually change anything user-facing, so I've bumped only the `PATCH` number.

File: docs/src/changelog.md [hunk 2]
```diff
@@ -7,6 +7,25 @@ CurrentModule = JuMP
...
+ - Fixed printing `MOI.Interval` with `MIME"text/latex"` (#3866)
+
+### Other
+
+ - Various minor improvements to the documentation (#3855) (#3860)
+ - Added [MathOptAI.jl](@ref) and [MathOptSymbolicAD.jl](@ref) (#3858)
+ - Clarified [`add_to_expression!`](@ref) can add two expressions (#3859)
+ - Added `SHOT` to the installation table (#3853)
+ - Improvements to test coverage (#3867) (#3868) (#3869) (#3870) (#3871) (#3872)
+   (#3873) (#3874) (#3875)
```
Comment (odow) [inline docs/src/changelog.md:25]: We now have >99.5% code coverage:

image

File: docs/src/changelog.md [hunk 3]
```diff
@@ -7,6 +7,25 @@ CurrentModule = JuMP
...
+### Fixed
+
+ - Fixed `UnsupportedNonlinearOperator` error for the single argument
+   `LinearAlgebra.norm` (#3864)
+ - Fixed printing `MOI.Interval` with `MIME"text/latex"` (#3866)
+
+### Other
+
+ - Various minor improvements to the documentation (#3855) (#3860)
+ - Added [MathOptAI.jl](@ref) and [MathOptSymbolicAD.jl](@ref) (#3858)
```
Comment (blegat) [inline docs/src/changelog.md:21]:
```suggestion
 - Added [MathOptAI.jl](@ref) and [MathOptSymbolicAD.jl](@ref) to the list of extensions in the documentation (#3858)
```

File: docs/src/changelog.md (hunk 1 above)
Comment (blegat) [inline docs/src/changelog.md:27]:
```suggestion
   variable with bounds for improving model creation performance with some solvers (#3863) (#3865)
```

**Review by blegat — APPROVED** [review]

**Pull request merged**